src/signals/incident/components/form/Header/index.test.js

```diff
@@ -1,93 +1,122 @@
+
+    expect(queryByText('(optioneel)')).not.toBeInTheDocument();
+  });
+
+  it('should render subtitle', () => {
+    const subtitle = 'Bar bazzz';
+    const { getByText } = render(withAppContext(<Header hasError={() => {}} meta={{ subtitle }} />));
+
+    expect(getByText(subtitle)).toBeInTheDocument();
+  });
+
+  it('should render children', () => {
+    const { container } = render(
+      withAppContext(
+        <Header hasError={() => {}}>
+          <span className="child" />
+        </Header>
+      )
     );
+
+    expect(container.querySelector('.child')).toBeInTheDocument();
   });
 
-  it('should render correctly', () => {
-    expect(wrapper).toMatchSnapshot();
+  it('should render required error', () => {
+    const hasError = prop => prop === 'required';
+
+    const { queryByText, rerender } = render(withAppContext(<Header hasError={() => false} touched />));
+
+    expect(queryByText('Dit is een verplicht veld')).not.toBeInTheDocument();
+
+    rerender(withAppContext(<Header hasError={hasError} touched />));
+
+    expect(queryByText('Dit is een verplicht veld')).toBeInTheDocument();
   });
 
-  it('should render when label is not available', () => {
-    wrapper.setProps({
-      meta: {
-        label: null,
-      },
-    });
+  it('should render email error', () => {
+    const hasError = prop => prop === 'email';
+
+    const { queryByText, rerender } = render(withAppContext(<Header hasError={() => false} touched />));
+
+    expect(queryByText('Het moet een geldig e-mailadres zijn')).not.toBeInTheDocument();
+
+    rerender(withAppContext(<Header hasError={hasError} touched />));
 
-    expect(wrapper).toMatchSnapshot();
+    expect(queryByText('Het moet een geldig e-mailadres zijn')).toBeInTheDocument();
   });
 
-  describe('error messages', () => {
-    it('should render no error when element is only touched', () => {
-      wrapper.setProps({ touched: true });
+  it('should render maxLength error', () => {
+    const requiredLength = 300;
+    const hasError = prop => prop === 'maxLength';
+    const getError = () => ({ requiredLength });
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    const { queryByText, rerender } = render(withAppContext(<Header hasError={() => false} getError={getError} touched />));
 
-    it('should render required error', () => {
-      hasError.mockImplementation(type => type === 'required');
-      wrapper.setProps({ touched: true });
+    expect(queryByText(`U kunt maximaal ${requiredLength} tekens invoeren.`)).not.toBeInTheDocument();
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    rerender(withAppContext(<Header hasError={hasError} getError={getError} touched />));
 
-    it('should render email error', () => {
-      hasError.mockImplementation(type => type === 'email');
-      wrapper.setProps({ touched: true });
+    expect(queryByText(`U kunt maximaal ${requiredLength} tekens invoeren.`)).toBeInTheDocument();
```

Comment:
You could define the error message string in a `const` before, so you're absolutely sure you use the same string in both `expect` statements.

Reply:
I thought about that, but it's only two occurrences 🤷‍♂️

Reply:
I know, but if you make a typo in the negative (`not`) test case, you won't notice it while your test isn't testing properly. So, still valuable, even if it's only two occurrences.

Reply:
Done